src/components/InputField/InputField.tsx

```diff
@@ -0,0 +1,87 @@
+import { COLORS } from "@src/constants/colors";
+import { useEffect, useState } from "react";
+import { TextInput, View, TouchableOpacity } from "react-native";
+
+import { styles, inputStyles } from "./InputField.styles";
+import { InputFieldProps } from "./InputField.types";
+import { ErrorIcon } from "./assets/ErrorIcon";
+import { PasswordHideIcon } from "./assets/PasswordHideIcon";
+import { PasswordShowIcon } from "./assets/PasswordShowIcon";
+import { ValidIcon } from "./assets/ValidIcon";
+import { Typography } from "../Typography";
+
+export const InputField = ({
+  initVal = "",
+  getVal,
+  errorConditions = false,
+  errorChildren = "Textfield is invalid",
+  password = false
+}: InputFieldProps) => {
+  const [textVal, setTextVal] = useState(initVal);
+  const [isActive, setIsActive] = useState(false);
+  const [hasError, setHasError] = useState(errorConditions);
+  const [isPassword] = useState(password);
+  const [showPwd, setShowPwd] = useState(false);
+
+  useEffect(() => {
+    getVal(textVal);
+  }, [textVal]);
+
+  useEffect(() => {
+    setHasError(errorConditions);
+  }, [errorConditions]);
+
+  const handleFocus = () => {
+    setIsActive(true);
+  };
+
+  const handleBlur = () => {
+    setIsActive(false);
+  };
+
+  const handlePress = (value: boolean) => {
+    setShowPwd(value);
```

Comment:
I believe we do not need to get parameter here.

```suggestion
    setShowPwd(prev => !prev);
```